doc/source/tutorial/stats/sampling_tdr.rst

```diff
@@ -0,0 +1,90 @@
+
+.. _sampling-tdr:
+
+Transformed Density Rejection (TDR)
+===================================
+
+.. currentmodule:: scipy.stats
+
+* Required: T-concave PDF, dPDF
+* Optional: mode, center
+* Speed:
+
+  * Set-up: slow
+  * Sampling: fast 
+
+
+TDR is an acceptance/rejection method that uses the concavity of a transformed
+density to construct hat function and squeezes automatically. Such PDFs are
+called T-concave. Currently the following transformations are implemented:
+
+.. math::
+
+    c = 0.: T(x) &= \log(x)\\
+    c = -0.5: T(x) &= \frac{1}{\sqrt{x}} \text{ (Default)}
+
+In addition to the PDF, it also requires the derivative of the PDF w.r.t x
+(i.e. the variate). These functions must be present as methods of a python
+object which can then be passed to the generators to instantiate their object.
+
+Three variants of this method are available:
+
+* GW: squeezes between construction points.
+* PS: squeezes proportional to hat function. (Default)
+* IA: same as variant PS but uses a composition method with
+  "immediate acceptance" in the region below the squeeze.
+
+This can be changed by passing a ``variant`` parameter.
+
+An example of using this method is shown below:
+
+    >>> from scipy.stats import TransformedDensityRejection
+    >>> 
+    >>> class Distribution:
+    ...     def pdf(self, x):
+    ...         if abs(x) <= 1:
+    ...             return 1 - x*x
+    ...         return 0
+    ...     def dpdf(self, x):
+    ...         if abs(x) <= 1:
+    ...             return -2*x
+    ...         return 0
+    ... 
+    >>> dist = Distribution()
+    >>> 
+    >>> urng = np.random.default_rng(0xbb51a838c8a087854ad3643b2b268d43)
+    >>> rng = TransformedDensityRejection(dist, seed=urng)
+    >>> rng.rvs()
+    0.39606414669189344
+
```

Comment:
i suggest to add how to use the method sqhratio and explain what that number means, I think it is in the Runuran vignette, I can double-check.

Reply:
> i suggest to add how to use the method sqhratio and explain what that number means

Sure. There are also a couple of other attributes. I will add an explanation for each of them.

> I think it is in the Runuran vignette, I can double-check

Thanks! That would be helpful.

Reply:
I have added a detailed explanation of the `sqhratio` attribute.

Reply:
great, I need to review it in more detail and I added a suggestion below